apluslms-yamlidator/apluslms_yamlidator/utils/collections.py

```diff
@@ -93,44 +93,53 @@ def data_updated(self, key, data):
     def get_data(self):
         return self._data
 
+    @property
+    def top_parent(self):
+        if not self._parent:
+            return self
+        parent = self._parent
+        if not parent:
+            return self
```

Comment:
:thinking: `if not self._parent and not self._parent`, I think you have the same tests twice.